src/js/components/service-requests/service-request-map.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * Set or update the location of service request output and hidden field.
+ * @param {array} coordinates - the lat and lng as an array
+ */
+function setSelectedLocation(coordinates) {
+  const $serviceRequestLocationPicker = $("#service-request-location-picker");
+  const $coordinatesHiddenField = $("#service-request-coordinates");
+  const $coordinatesOutput =
+    $serviceRequestLocationPicker &&
+    $serviceRequestLocationPicker.find(".map-coordinates");
+  const $removeCoordinatesButton = $serviceRequestLocationPicker.find(
+    ".button--remove-location"
+  );
+
+  if ($coordinatesOutput) {
+    $coordinatesOutput.text(coordinates.join(", "));
+  }
+
+  if ($coordinatesHiddenField) {
+    $coordinatesHiddenField.val(coordinates.join(", "));
+  }
+
+  if ($removeCoordinatesButton) {
+    $removeCoordinatesButton.removeClass("button--disabled");
+    $removeCoordinatesButton.on("click", (event) => {
+      event.preventDefault();
+
+      $coordinatesHiddenField.val("");
+      $coordinatesOutput.text("No location selected");
+      $removeCoordinatesButton.addClass("button--disabled");
+      $(this).off(event);
+    });
+  }
+}
+
+/**
+ * Called when the user clicks the "Use my location" button. Get's the
+ * users current location and updates the location of your service
+ * request output field. Also sets the associated hidden form field.
+ * @param {object} map - the current map instance
+ */
+function getLocation(map) {
+  map.locate({ setView: true });
+
+  map.on("locationerror", (error) => console.error(error));
+
+  map.on("locationfound", (location) => {
+    const coordinates = [location.latitude, location.longitude];
+    map.setView(coordinates, 13);
+    setSelectedLocation(coordinates);
```

Comment:
this makes strong assumptions about this component only being used once on the page. I don't see when that wouldn't be the case, but the idea with component classes is that each instance of a component is totally independent from assumptions about what else is on the page - it would be better design for these to be instance methods so you'd call `this.setSelectedLocation`

Reply:
Makes sense.